src/core/config/border_router.h

```diff
@@ -65,6 +65,16 @@
 #define OPENTHREAD_CONFIG_BORDER_ROUTING_ENABLE 0
 #endif
 
+/**
+ * @def OPENTHREAD_CONFIG_BORDER_ROUTING_MAX_DISCOVERED_PREFIXES
+ *
+ * Specifies maximum number of discovered prefixes (on-link prefixes on the infra link) maintained by routing manager.
+ *
+ */
+#ifndef OPENTHREAD_CONFIG_BORDER_ROUTING_MAX_DISCOVERED_PREFIXES
+#define OPENTHREAD_CONFIG_BORDER_ROUTING_MAX_DISCOVERED_PREFIXES 8
```

Comment:
Should we define `OPENTHREAD_POSIX_CONFIG_MAX_EXTERNAL_ROUTE_NUM` to `OPENTHREAD_CONFIG_BORDER_ROUTING_MAX_DISCOVERED_PREFIXES` by default? 

@superwhd

Reply:
That sounds reasonable to me but I think should be done in a new PR (for a posix related change).